File: JAVA/level1/최대공약수와_최소공배수.java
class Solution {

    public int gcd(int n, int m){

        if(n < m){
            int temp = m;
            m = n;
            n = temp;
        }

        if(m == 0)
            return n;
        return gcd(m, n - (n/m)*m);
    }
    public int[] solution(int n, int m) {
        int[] answer = new int[2];

        answer[0] = gcd(n, m);
        answer[1] = n*m/gcd(n, m);

        return answer;
    }
}

File: JavaScript/level1/최대공약수와_최소공배수.js
function gcd(a, b) {
    let temp;
    
    while(b !== 0){
        temp = b;
        b = a%b;
        a = temp;
    }
    return a;
};

function solution(n, m) {
    var answer = [];
    
    if(n > m)
        answer.push(gcd(n, m));
    else
        answer.push(gcd(m, n));
        
    answer.push(answer[0]*(m/answer[0])*(n/answer[0]));
    
    return answer;
}